apiserver/common/addresses.go

```diff
@@ -69,7 +83,7 @@ func (api *APIAddresser) APIAddresses() (params.StringsResult, error) {
 }
 
 func apiAddresses(getter APIHostPortsGetter) ([]string, error) {
-	apiHostPorts, err := getter.APIHostPorts()
+	apiHostPorts, err := getter.APIHostPortsForAgents()
```

Comment:
note that "getter" needs to change as you have done here (because that would go directly to the state backend).